src/token.rs

```diff
@@ -278,35 +238,30 @@ where
     where
         I: IntoIterator<Item = (S, Option<T>)>,
     {
-        let feature_map =
+        let features =
             BTreeMap::from_iter(iter.into_iter().map(|(k, v)| (k.into(), v.map(Into::into))));
-        let features = map_to_string(&feature_map);
-
-        let lazy_feature_map = Lazy::new();
-        lazy_feature_map.get_or_create(|| feature_map);
 
-        Features {
-            features,
-            feature_map: lazy_feature_map,
-        }
+        Features { inner: features }
     }
 }
 
-impl PartialEq for Features {
-    fn eq(&self, other: &Features) -> bool {
-        self.feature_map.get_or_create(|| self.as_map_eager())
-            == other.feature_map.get_or_create(|| other.as_map_eager())
+impl From<Features> for String {
+    fn from(features: Features) -> Self {
+        (&features).into()
     }
 }
 
-fn map_to_string(feature_map: &BTreeMap<String, Option<String>>) -> String {
-    feature_map
-        .iter()
-        .map(|(k, v)| match *v {
-            Some(ref v) => format!("{}:{}", k, v),
-            None => k.to_owned(),
-        })
-        .join("|")
+impl From<&Features> for String {
```

Comment:
Wouldn't `ToString` be more expected here?

Reply:
`ToString` is already implemented through `Display`. Another question is why we need this if we have `Display` ;).

Reply:
Well, then... 

I was only looking for implementors of `ToString` expecting to find something like `impl<T> ToString for T where String: From<&T>`, didn't even bother to check for a `Display` impl on `Token` ;)